server/controllers/admin.py

```diff
@@ -124,6 +126,17 @@ def assignment(cid, aid):
     if assgn.course != current_course:
         return abort(401)
 
+    assgn.name = form.name._value()
```

Comment:
What happens if it's not a POST (or a valid submission?) - We want to validate that they submitted the form (and that the forms are valid). 

1. Method is POST. 
2. form.validate_on_submit()  (before you create the form from `assgn`) 

Take a look at `new_assignment`.